packages/core/src/customer/strategies/stripe-upe/stripe-upe-customer-strategy.ts

```diff
@@ -0,0 +1,119 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { PaymentMethodActionCreator } from '../../../payment';
+import { StripeElements, StripeElementType, StripeScriptLoader, StripeUPEClient, StripeEventType, StripeUPEAppearanceOptions } from '../../../payment/strategies/stripe-upe';
+import CustomerActionCreator from '../../customer-action-creator';
+import CustomerCredentials from '../../customer-credentials';
+import { CustomerInitializeOptions, CustomerRequestOptions, ExecutePaymentMethodCheckoutOptions } from '../../customer-request-options';
+import CustomerStrategy from '../customer-strategy';
+
+export default class StripeUPECustomerStrategy implements CustomerStrategy {
+    private _stripeElements?: StripeElements;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _stripeUPEScriptLoader: StripeScriptLoader,
+        private _customerActionCreator: CustomerActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator
+    ) {}
+
+    async initialize(options: CustomerInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId, stripeupe: { container, onEmailChange, getStyles, isLoading } = {} } = options;
+        let stripeUPEClient: StripeUPEClient;
+
+        if (!methodId) {
+            throw new InvalidArgumentError('Unable to proceed because "methodId" argument is not provided.');
+        }
+
+        const {
+            paymentMethods: { getPaymentMethodOrThrow }, customer: { getCustomerOrThrow },
+        } = await this._store.dispatch(
+            this._paymentMethodActionCreator.loadPaymentMethod(methodId, { params: { method: 'card' } })
+        );
+        const {
+            clientToken, initializationData: { stripePublishableKey, stripeConnectedAccount } = {}
+        } = getPaymentMethodOrThrow('card', methodId);
```

Comment:
```
getPaymentMethodOrThrow(
        methodId: string,
        gatewayId?: string
    )
```
Seems like we are passing methodId as gatewayId, can we please address that since that can be confusing.

Reply:
Hey @animesh1987, I did that because CustomerRequestOptions and ShippingRequestOptions only accept `methodId` but I totally understand can be confusing, I created a commit called `feat(payment): STRIPE-52 animesh & peng a proposal to solve the feedback` to handle a possible solution for that, let me know if It's ok and tomorrow I would incorporate the JS part on the JS PR

Reply:
Thanks @AlanMunozN that is much better :D